moshi/src/test/java/com/squareup/moshi/internal/RecursiveTypesResolveTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2017 Gson Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.squareup.moshi.internal;
```

Comment:
I'm not entirely sure this should be in the `internal` package because it's testing changes made in `Types`, which is not in the `internal` package.

Reply:
Ahh good shout 37aaaae